### ISSUE TITLE
Option to create internal Wiki

### ISSUE THREAD

**Issue by blewsky** [frappe/wiki#60]
It would be very beneficial if (parts of) the Wiki can be protected password. This would allow the creation of an internal wiki that could be used e.g. by employees and which isn't seen by outsiders. Currently (if I understand the wiki correctly) the wiki is automatically open to access by anyone with access to the link.

**Potential solution:**
A check box could be added that restricts the respective page to internal use. If it is activated one could not see the page without being logged in into Frappe.

**Potential conflicts:**
There are different roles on ERPnext/Frappe that allow external users (e.g. auditors to get access to some selected parts of ERPnext/Frappe. It could create a problem, if these external users could access the internal wiki. So it would be probably good to have restrictions on the Roles who might view the internal wiki and who not.

**Comment by hasnain2808** [comment]
Hi @blewsky 
Wiki can be used as a Internal Wiki
We are using it 
For any wiki page if you turn off `allow guest`, it will be password protected
![image](https://user-images.githubusercontent.com/28212972/145195131-f78aa09f-703b-436d-987e-4218d68eb029.png)

Just now I also added a small thing that a new page would inherit the value of `allow guest` from the page on which new page was clicked

**Comment by hasnain2808** [comment]
A better Role based system could be really nice though

**Comment by blewsky** [comment]
Exactly. More specific role based permissions would be critical if the wiki was used for internal/sensitive information. You wouldn't want to have an auditor or customer accessing your internal wiki.

**Comment by hans-permana** [comment]
Also another use case is where a certain information is only available for Finance team, or leadership team, and not the entire company member. Linking this with Frappe Role Permission would really be awesome.

**Comment by blewsky** [comment]
Is there somehow a way to use the role-based access control used in ERPnext for the wiki? I am not familiar with the internal way how the RBAC is implemented for ERPnext. If it is something in the Frappe framework, one could probably extend it to the wiki more easily to assign roles etc. that are allowed to read/edit the wiki. 

So far the lack of the RBAC is the dealbreaker for us why we can't use the wiki. Would be great to see this implemented soon. I guess without proper RBAC, the wiki is probably only of limited us as an internal knowledge base in an ERP setting (due to the above mentioned reasons by me and others).

**Comment by Ragevn** [comment]
hey, this exactly what i am looking for. currently wiki version-14 have a problem that i can access the wiki page without login even when i'm not allow guest user. sometime it redirect to login page and sometime it doesn't. and that lead to serious problem that i can't put some important docs on wiki. i tried to use Basic Authentication but the wiki page log me out with a expired session message. any solution that i can keep my wiki page  private?

**Comment by lamphanjk** [comment]
yes, It's the exactly feature what we want. 
We want use wiki for internal but must be limited by  group of users or Department. So Hope we can define role/ permission for page level for Wiki. 

Hope someone can support on this soon. 
best regards,

**Comment by BreadGenie** [comment]
I'll take a stab at this when I get some time

What I'm thinking of is restriction to be added in `Wiki Space` and not `Wiki Page` as it makes more sense (set of pages are usually shared among the team instead of a single page)

**Comment by lamphanjk** [comment]
So, could I know any update on role/ permission with WIki, I love it but can not use if we can not control permission. 
Hope to have a good new from you.
Thanks

**Comment by nilpatel42** [comment]
is any one is working on this feature ?

**Comment by cdiethelm** [comment]
Definitely interested in this feature too! Would be great to have permissions on a wiki site for the different departments of a company. Also different permissions on who can read or write to the wiki.

**Comment by SalmanShamsKhan** [comment]
Assalam oalikum everyone,

In order to make wiki usefull for internal documentation, you need to uncheck 'allow guest to view" on doctype level of "wiki page"
![Screenshot 2023-12-22 004323](https://github.com/frappe/wiki/assets/85616727/227844fb-03ca-493a-abd4-264bc45d16f7)

then also while making a "wiki page", untick the allow guest on the wiki page
![Screenshot 2023-12-22 004707](https://github.com/frappe/wiki/assets/85616727/936d65b4-2a1f-4307-9471-1709596a104c)


Now if you try to access the page, it will only open via login
**Note: before making your doctype changes, switch the site to develops mode
"developer_mode": 1
add this line to the site_config.json save it so that the site can be developers mode**

Have a great day hustlers, Happy Learning :)

**Comment by monolithonadmin** [comment]
We have started working on a solution, hopefully it will be merged soon. @BreadGenie could you take a look into this:
https://github.com/frappe/wiki/pull/217/commits/87957daaa30fdba25e8e2a641142df51bf9b3b43

Thank you

**Comment by blewsky** [comment]
@backendoadmin Any updates on this?

**Comment by NagariaHussain** [comment]
One simple way to do this would be to have a role list (Table Multi Select) in Wiki space to restrict the Wiki Space to those particular roles.

**Comment by techyaar** [comment]
I found a practical **Wiki V3 workaround for internal/private wiki spaces**, and after deeper testing there’s an important nuance.

For anyone trying to make a wiki space internal while preserving the **published V3 reading experience**:

### ✅ Required setup

1. Keep the **Wiki Space Published = ON**

   * this keeps the route alive
   * preserves Mermaid / markdown rendering
   * keeps the sidebar + navigation tree
   * maintains the full V3 docs UX

2. Go to **Desk → Wiki Document → Tree**

   * open your wiki space tree
   * example: `LEGO [Root Group]`

3. Mark **every Wiki Document in the tree** as:

   * `Is Private = ON`
   * `Is Published = ON`

### ⚠️ Important Wiki V3 behavior

Initially I assumed marking only the **Root Group as Private** was enough, but direct child routes like:

`/lego/home`

were still publicly accessible in incognito.

After testing, the route resolver appears to check the **individual Wiki Document matched by the URL**, not strict inherited privacy from the ancestor tree.

So the working solution is:

> **set every folder + child page + leaf document to Private individually**

Once all documents in the tree are private, the published V3 wiki keeps the full internal documentation UX while correctly forcing Frappe authentication for guests.

This preserves:

* proper rendering
* Mermaid
* navigation tree
* markdown workflows
* sidebar hierarchy
* beautiful published docs UX
* internal login protection

Tested successfully on **Wiki V3 with Frappe/ERPNext v16**.

**Comment by adamBoualleiguie** [comment]
Hello , thanks for such cool app <3 
it would be great if wiki app manages privates spaces , public  spaces , also private spaces allowed only for set of users or teams internally
